OpenRA.Mods.Common/Widgets/CheckboxWidget.cs

```diff
@@ -54,17 +56,21 @@ public override void Draw()
 			var disabled = IsDisabled();
 			var font = Game.Renderer.Fonts[Font];
 			var color = GetColor();
+			var hover = Ui.MouseOverWidget == this;
 			var colordisabled = GetColorDisabled();
 			var bgDark = GetContrastColorDark();
 			var bgLight = GetContrastColorLight();
 			var rect = RenderBounds;
 			var text = GetText();
 			var textSize = font.Measure(text);
 			var check = new Rectangle(rect.Location, new Size(Bounds.Height, Bounds.Height));
-			var baseName = IsHighlighted() ? "checkbox-highlighted" : "checkbox";
-			var state = WidgetUtils.GetStatefulImageName(baseName, disabled, Depressed && HasPressedState, Ui.MouseOverWidget == this);
 
-			WidgetUtils.DrawPanel(state, check);
+			if (HasBackground)
+			{
+				var baseName = IsHighlighted() ? "checkbox-highlighted" : "checkbox";
+				var state = WidgetUtils.GetStatefulImageName(baseName, disabled, Depressed, hover);
+				WidgetUtils.DrawPanel(state, check);
```

Comment:
The logic here is redundant with `DrawBackground` method inherited from `ButtonWidget`. So I suggest using it instead:

```csharp
if (HasBackground)
	DrawBackground(Background, check, disabled, Depressed, hover, IsHighlighted());
```

But for that to be useful a `Background` property needs to be added in the checkbox widget:
```csharp
public new string Background = "checkbox";
```

Reply:
Now that we define `Background` we don't really need `HasBackground`. The `DrawBackground` method already has a method for checking wether the string is empty